src/main/java/seedu/duke/manager/Manager.java

```diff
@@ -0,0 +1,103 @@
+package seedu.duke.manager;
+
+import seedu.duke.command.BadCommandException;
+import seedu.duke.command.Command;
+import seedu.duke.command.CommandParser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Manager is the superclass for all WellNUS++ event drivers <br>
+ * <br>
+ * Each manager is in charge of 'managing' exactly one feature. <br>
+ * For example, <i>hb</i> and <i>reflect</i>. <br>
+ * <br>
+ * The manager should run an event driver (infinite loop) and is in charge
+ * of a Feature's input, output, 'business' logic and graceful termination
+ */
+public abstract class Manager {
+
+    private final CommandParser commandParser;
+    private ArrayList<Command> argumentList;
+
+    /**
+     * Construct a feature Manager to handle control flow for the given feature.
+     */
+    public Manager() {
+        this.commandParser = new CommandParser();
+    }
+
+    /**
+     * Utility function to get the CommandParser tied to the Manager class.
+     *
+     * @return commandParser reference to this manager's instance of CommandParser
+     */
+    public CommandParser getCommandParser() {
+        return this.commandParser;
+    }
```

Comment:
Just checking if there is a need for WellNUS or other managers to get this commandParser

Reply:
The idea is that the control flow logic will be done in the `run()` event driver  

I anticipate that the control flow logic will include IO for all Manager instances and subclasses. Parsing of the input would naturally require the CommandParser, so I included this getter

If a custom `CommandParser` needs to be used, this function can be overriden

What are your views?

Reply:
I agree with Nicholas. My concept of the command workflow would be having Manager call the `Ui` class to read input(commands), parse command(`CommandParser`), check which `Command` class to delegate to and call `Command.execute()`. So Manager is literally managing the entire feature, although most of its work is delegated to other classes.

What do yall think? @wenxin-c @nichyjt @YongbinWang

Reply:
Makes sense!